src/models/connectionCredentials.ts

```diff
@@ -204,7 +205,10 @@ export class ConnectionCredentials implements IConnectionCredentials {
                     }
                     return undefined;
                 },
-                onAnswered: (value) => credentials.password = value
+                onAnswered: (value) => {
+                    credentials.password = value;
+                    (<IConnectionProfile>credentials).emptyPasswordInput = utils.isEmpty(credentials.password);
```

Comment:
Is this type assertion guaranteed to be true? If so, I think we should just update the method to take a IConnectionProfile object. If not, this code should probably check if credentials is actually an IConnectionProfile object.